worlds/lufia2ac/basepatch/basepatch.asm

```diff
@@ -0,0 +1,869 @@
+pullpc
+
+LoadRX:
+    JSL $809099             ; (overwritten instruction) load save slot A from SRAM
+    SEP #$10
+    REP #$20
+    ASL
+    ASL
+    TAX
+    LDA $F027E0,X           ;
+    STA $F02800             ; restore received count
+    LDA $F027E2,X           ;
+    STA $F02802             ; restore processed count
+    SEP #$20
+    REP #$10
+    RTL
+
+
+
+; keep inventory after defeat
+pushpc
+org $848B9C
+    NOP #5                  ; overwrites LDA.b #$FF : STA $7FE759 : JSR $8888
+    JSL DeathLinkTX
+pullpc
+
+
+
+; set initial floor number
+pushpc
+org $8487A9
+    JSL InitialFloor        ; overwrites TDC : STA $7FE696
+    NOP
+pullpc
+
+InitialFloor:
+    LDA $D08015             ; read initial floor number
+    STA $7FE696             ; (overwritten instruction)
+    TDC                     ; (overwritten instruction)
+    RTL
+
+
+
+; report final floor goal completion
+pushpc
+org $839E87
+    JSL FinalFloor          ; overwrites STA $0005B0
+pullpc
+
+FinalFloor:
+    STA $0005B0             ; (overwritten instruction)
+    LDA.b #$01
+    STA $F02034             ; report final floor goal
+    RTL
+
+
+
+; start with Providence
+pushpc
+org $8488BB
+    SEC                     ; {carry clear = disable this feature, carry set = enable this feature}
+    JSL Providence          ; overwrites LDX.w #$1402 : STX $0A8D
+    NOP                     ;
+pullpc
+
+Providence:
+    LDX.w #$1402            ; (overwritten instruction)
+    STX $0A8D               ; (overwritten instruction) add Potion x10
+    BCC +
+    LDX.w #$022D            ;
+    STX $0A8F               ; add Providence
++:  RTL
+
+
+
+; start inventory
+pushpc
+org $848901
+    JSL StartInventory      ; overwrites JSL $81ED35
+pullpc
+
+StartInventory:
+    JSL $81ED35             ; (overwritten instruction)
+    REP #$20
+    LDA $F02802             ; number of items to process
+    DEC
+    BMI ++                  ; skip if empty
+    ASL
+    TAX
+-:  LDA $F02804,X           ; item ID
+    BPL +                   ; spells have high bit set
+    PHX
+    JSR LearnSpell
+    PLX
++:  BIT.w #$C000            ; ignore blue chest items (and spells)
+    BNE +
+    PHX
+    STA $09CF               ; specify item ID
+    TDC
+    INC
+    STA $09CD               ; specify quantity as 1
+    JSL $82E80C             ; add item to inventory
+    REP #$20
+    PLX
++:  DEX
+    DEX
+    BPL -
+++: SEP #$20
+    RTL
+
+
+
+; increase variety of red chest gear after B9
+pushpc
+org $839176
+    CLC                     ; {carry clear = disable this feature, carry set = enable this feature}
+    JSL RedChestGear        ; overwrites LDX.w #$1000 : LDA $60
+org $83917D
+    JSL RunEquipmentRNG     ; overwrites LSR : JSR $9E11
+pullpc
+
+RedChestGear:
+    BCC +
+    REP #$20                ; support more than 127 items
++:  LDX.w #$1000            ; (overwritten instruction)
+    LDA $60                 ; (overwritten instruction)
+    RTL
+RunEquipmentRNG:
```

Comment:
I see a write to $E746,Y within this function, which suggests that DB is either $7E or $7F. This would be good to document.